source/FreeRTOS_IP_Utils.c

```diff
@@ -983,18 +983,22 @@ void vPreCheckConfigs( void )
 
     #if ( configASSERT_DEFINED == 1 )
     {
-        volatile size_t uxSize = sizeof( uintptr_t );
-
-        if( uxSize == 8U )
-        {
-            /* This is a 64-bit platform, make sure there is enough space in
-             * pucEthernetBuffer to store a pointer and also make sure that the value of
-             * ipconfigBUFFER_PADDING is such that (ipconfigBUFFER_PADDING + ipSIZE_OF_ETH_HEADER) is a
-             * 32 bit (4 byte) aligned value, so that when incrementing the ethernet buffer with
-             * (ipconfigBUFFER_PADDING + ipSIZE_OF_ETH_HEADER) bytes it lands in a 32 bit aligned address
-             * which lets us efficiently access 32 bit values later in the packet. */
-            configASSERT( ( ipconfigBUFFER_PADDING >= 14 ) && ( ( ( ( ipconfigBUFFER_PADDING ) + ( ipSIZE_OF_ETH_HEADER ) ) % 4 ) == 0 ) );
-        }
+        size_t uxSize;
+
+        #if ( UINTPTR_MAX > 0xFFFFFFFFU )
+
+            /*
+             * This is a 64-bit platform, make sure there is enough space in
+             * pucEthernetBuffer to store a pointer.
+             */
+            configASSERT( ipBUFFER_PADDING >= 14U );
+        #else
+            /* This is a 32-bit platform. */
+            configASSERT( ipBUFFER_PADDING >= 10U );
+        #endif /* UINTPTR_MAX > 0xFFFFFFFFU */
+
+        /* And it must have this strange alignment: */
```

Comment:
Can we add that the purpose of this assert is to make sure that the contents of the network packets after the ethernet header lands in a 32 bit aligned address as a comment?

I think, changing the assert to: `configASSERT( ( ( ( ipBUFFER_PADDING ) + ( ipSIZE_OF_ETH_HEADER ) ) % 4U ) == 0U );` would also make it more clear to the reader.

Reply:
We have explained it already in other locations. I added a short comment to the asserts:
~~~diff
+    /* Check if ipBUFFER_PADDING has a minimum size, depending on the platform.
+     * See FreeRTOS_IP.h for more details. */
~~~